src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -1646,6 +1646,12 @@ public async Task Step(int threadId, enum_STEPKIND kind, enum_STEPUNIT unit)
                         await MICommandFactory.ExecStep(threadId);
                         break;
                     case enum_STEPKIND.STEP_OVER:
+                        string line = await ConsoleCmdAsync("f", false, true);
+                        if (line.Contains("return "))
```

Comment:
Simple heuristic that works well in practice.
Unfortunately there's no easy way to catch stepping over function calls.

Reply:
I don't think I like this idea. It seems like it might be better to just tell users that if they step out they will get this feature.

If others disagree, I think at the least --
1. You want to change this to instead just checking if the line has "return" anywhere in it you should instead check: `^[\s\{]*return\s`. Otherwise stepping over `if (false) return 27;` would result in a step out.
2. This needs to be behind an off-by-default option

Reply:
Yeah it was more of a test.
In the end it doubles the commands run for every step. This really needs gdb support.
https://sourceware.org/bugzilla/show_bug.cgi?id=26354